sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1839,13 +1841,18 @@ class Analyzer(
                   }
                   AggregateExpression(agg, Complete, isDistinct, filter)
                 // This function is not an aggregate function, just return the resolved one.
-                case other =>
-                  if (isDistinct || filter.isDefined) {
-                    failAnalysis("DISTINCT or FILTER specified, " +
-                      s"but ${other.prettyName} is not an aggregate function")
-                  } else {
-                    other
+                case other if (isDistinct || filter.isDefined) =>
+                  failAnalysis("DISTINCT or FILTER specified, " +
+                    s"but ${other.prettyName} is not an aggregate function")
+                case e: StringTrim if arguments.size == 2 =>
+                  if (trimWarningEnabled.get) {
+                    log.warn("Two-parameter TRIM function signature is deprecated." +
```

Comment:
shall we list all of them? `TRIM` -> `TRIM/LTRIM/RTRIM`, `TRIM(trimStr FROM str)` -> `TRIM((BOTH | LEADING | TRAILING)? trimStr FROM str)`

Reply:
Got it. BTW, that will be the same with my first commit.

Reply:
My bad. I had to ask from the beginning to avoid my misunderstanding here.